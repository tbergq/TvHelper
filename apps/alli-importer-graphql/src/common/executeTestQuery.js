// @flow

import { graphql } from 'graphql';

import schema from '../Schema';

const createContext = () => ({});

export default function executeTestQuery(
  query: string,
  variables: ?{ ... },
  request: ?{ ... } = {},
): Promise<{ ... }> {
  // $FlowExpectedError: Ok for testing purposes
  return graphql(schema, query, null, createContext(request), variables);
}
