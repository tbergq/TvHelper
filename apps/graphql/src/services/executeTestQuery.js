// @flow

import { graphql } from 'graphql';

import schema from '../Schema';
import createContext from './createGraphqlContext';

export default function executeTestQuery(
  query: string,
  variables: ?{ ... },
  request: ?{ ... } = {},
): Promise<{ ... }> {
  // $FlowExpectedError: Ok for testing purposes
  return graphql(schema, query, null, createContext(request), variables);
}
