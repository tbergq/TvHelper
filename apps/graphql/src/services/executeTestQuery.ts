import { graphql } from 'graphql';

import schema from '../Schema';
import createContext from './createGraphqlContext';

export default function executeTestQuery(
  query: string,
  variables: {} | null | undefined,
  request: {} | null | undefined = {},
): Promise<{}> {
  // @ts-ignore: Ok for testing purposes
  return graphql(schema, query, null, createContext(request), variables);
}
