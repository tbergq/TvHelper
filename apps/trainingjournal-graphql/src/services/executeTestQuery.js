// @flow

import { graphql } from 'graphql';

import schema from '../Schema';

export default function executeTestQuery(query: string, variables: ?{ ... }): Promise<{ ... }> {
  return graphql(schema, query, null, null, variables);
}
