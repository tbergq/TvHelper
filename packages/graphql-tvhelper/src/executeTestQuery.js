// @flow

import { graphql as originalGraphQL, GraphQLSchema, GraphQLObjectType } from 'graphql';

import SearchTvShow from './search/queries/SearchTvShow';

const RootQuery = new GraphQLObjectType({
  name: 'RootQuery',
  description: 'Root Query',
  fields: {
    searchTvShow: SearchTvShow,
  },
});

const schema = new GraphQLSchema({
  query: RootQuery,
});

/**
 * Executes GraphQL query using our schema. This function should
 * be used only during testing because it uses faked context.
 */
export default function executeTestQuery(
  query: string,
  variables: ?{ ... },
  context: ?{ ... },
): Promise<Object> {
  return originalGraphQL(schema, query, null, context, variables);
}
