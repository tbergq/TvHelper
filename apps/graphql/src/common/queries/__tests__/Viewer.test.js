// @flow

import executeTestQuery from '../../../services/executeTestQuery';

it('renders Unautorized type', async () => {
  const data = await executeTestQuery(
    `query Viewer {
      viewer {
        __typename
      }
}`,
    {},
    { user: { app: undefined } },
  );
  expect(data).toMatchSnapshot();
});

it('renders TJViewer type', async () => {
  const data = await executeTestQuery(
    `query Viewer {
      viewer {
        __typename
      }
}`,
    {},
    { user: { app: 'trainingjournal' } },
  );
  expect(data).toMatchSnapshot();
});

it('renders TVHViewer type', async () => {
  const data = await executeTestQuery(
    `query Viewer {
      viewer {
        __typename
        ... on TvHelperViewer {
          username
        }
      }
}`,
    {},
    { user: { app: 'tvhelper', username: 'tito' } },
  );
  expect(data).toMatchSnapshot();
});
