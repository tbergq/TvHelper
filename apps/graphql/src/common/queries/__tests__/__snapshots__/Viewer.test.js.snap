// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TJViewer type 1`] = `
Object {
  "data": Object {
    "viewer": Object {
      "__typename": "TraningJournalViewer",
    },
  },
}
`;

exports[`renders Unautorized type 1`] = `
Object {
  "data": Object {
    "viewer": Object {
      "__typename": "Unauthorized",
    },
  },
}
`;
